### PR TITLE
Include topic in layer search issue 35

### LINKED ISSUE #35: Include topic in layer search -- Bug 5684
See email exchange 6/12/2020. Search results for the term "ARC Lands" don't include the map layers in the ARC Lands topic. Mike has adjusted the search indexer to include the parent topic name so these layers show up.

I noticed while verifying this that even prior to this change the ARC Lands topic is included in the results (with type=category) returned from SOLR, but not making its way through the UI to show it as a topic/category/service that can be loaded. This is due to SOLR returning type=["Category"] (as an array), and my code expecting it to be a simple string.

https://repo.fla-etat.org/bugzilla/show_bug.cgi?id=5684

### PR DESCRIPTION
**Pull request by wmbeers**
Adding search by topic to search string request passed to SOLR, and handling "category" response from search as an array or string. (SOLR was configured to return category as array, not a string; we may fix that, but the viewer can handle the response either way with this change.)

### CODE REVIEW

File: viewer/js/gis/dijit/LayerLoader.js [hunk 1]
```diff
@@ -792,12 +792,16 @@ function (declare, _WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin, Dialog
                 encodedSearchTerms +
                 '"^100+OR+description:"' +
                 encodedSearchTerms +
-                '"^50+OR+name:' +
+                '"^50+OR+topic:"' +
+                encodedSearchTerms +
+                '"^75+OR+name:' +
```
Comment (wmbeers) [inline viewer/js/gis/dijit/LayerLoader.js:797]: Adds topic search by whole term (quoted) with 75 boost.

File: viewer/js/gis/dijit/LayerLoader.js [hunk 2]
```diff
@@ -792,12 +792,16 @@ function (declare, _WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin, Dialog
...
                 encodedSearchTerms +
                 '+OR+longName:' +
                 encodedSearchTerms +
                 '+OR+description:' +
                 encodedSearchTerms +
+                '+OR+topic:' +
+                encodedSearchTerms +
```
Comment (wmbeers) [inline viewer/js/gis/dijit/LayerLoader.js:804]: Adds topic search by all words (unquoted) with no boost.

File: viewer/js/gis/dijit/LayerLoader.js [hunk 3]
```diff
@@ -807,7 +811,10 @@ function (declare, _WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin, Dialog
                 var resultDocs = JSON.parse(reply).response.docs;
                 var searchResults = [];
                 resultDocs.forEach(function (doc) {
-                    if (doc.type === 'category') {
+                    //this business of checking if it's an array is because SOLR returns the type as an array. I don't think it should do that,
+                    //and perhaps we can fix it, so I've added a test to see if it's an array with value 'category' as a work-around,
+                    //and a test for simple string match if/when we do fix SOLR.
+                    if (doc.type && (Array.isArray(doc.type) && doc.type.indexOf('category') >= 0) || doc.type === 'category') {
```
Comment (wmbeers) [inline viewer/js/gis/dijit/LayerLoader.js:817]: Deals with category being returned as an array.

**Review by mikekonikoff — APPROVED** [review]
Changes look good!

**Comment by wmbeers** [comment]
Change Request approved by Pete. See email 7/22/20 10:14 a.m.

**Pull request merged**